### PR TITLE
Feature/forgot password

### PR DESCRIPTION
**Pull request by DavMovsesyan2002**

### CODE REVIEW

**Review by rikarsen — APPROVED** [review]

**Review by rikarsen — CHANGES_REQUESTED** [review]
image

you have error in build

**Review by VahMats — DISMISSED** [review]

File: src/pages/reset-password/request-password.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      <p className="mb-5 text-base text-black-light">
+        {t(Translation.PAGE_RESET_PASSWORD_DESCRIPTION)}
+      </p>
+      <div className="mb-5 w-full">{children}</div>
+      <Button
+        size="fl"
+        type="submit"
+        disabled={isResetPasswordLoading}
+      >
+        {t(Translation.PAGE_RESET_PASSWORD_BUTTON)}
```
Comment (gor017) [inline src/pages/reset-password/request-password.tsx:24]: let's add loading on the button, I suggest customize loading case in Button component

File: src/pages/reset-password/verify-otp.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      <div className="mb-5 mt-5">
+        <p className="mb-5 text-base text-black-light">{t(Translation.PAGE_VERIFY_OTP_LABEL)}</p>
+        {children}
+      </div>
+      <Button
+        type="submit"
+        disabled={isResetPasswordLoading}
+      >
+        {t(Translation.PAGE_VERIFY_OTP_BUTTON)}
+      </Button>
```
Comment (gor017) [inline src/pages/reset-password/verify-otp.tsx:32]: also add loading here

**Review by gor017 — CHANGES_REQUESTED** [review]

**Review by gor017 — DISMISSED** [review]

File: src/components/IsLoading/index.tsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import React from 'react'
+import IsLoadingIcon from '@components/Icons/IsLoadingIcon'
+
+const IsLoading = () => (
+  <div className="... mr-3 h-7 w-7 animate-spin">
```
Comment (ArthurO2002) [inline src/components/IsLoading/index.tsx:5]: ... ????

File: src/pages/reset-password/index.tsx [hunk 4]
```diff
@@ -1,128 +1,96 @@
...
 
-      return
-    }
-
-    router.query.tab = RouterQueryTypes.verifyOtp
-    router.push(router)
-    setEmailVerificationError('')
-  }
+  const onSubmit = (data: IResetPasswordParams) => {
+    methods.clearErrors()
```
Comment (ArthurO2002) [inline src/pages/reset-password/index.tsx:38]: Why are we clearing errors, if there is no error they will disappear

File: src/pages/reset-password/index.tsx [hunk 5]
```diff
@@ -1,128 +1,96 @@
...
+          {
+            email: data.email,
+            otp: data.otp?.toString(),
+          },
+          selectedIndex
+        )
+      )
+    } else {
+      dispatch(
+        usersMiddleware.resetPassword({
```
Comment (ArthurO2002) [inline src/pages/reset-password/index.tsx:54]: It's wrong each button should do it's own job

File: src/pages/reset-password/verify-otp.tsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+      </div>
+      <div className="mb-5 mt-5">
+        <p className="mb-5 text-base text-black-light">{t(Translation.PAGE_VERIFY_OTP_LABEL)}</p>
+        {children}
+      </div>
+      <Button
+        type="submit"
+        disabled={isResetPasswordLoading}
+      >
+        {isResetPasswordLoading ? <IsLoading /> : t(Translation.PAGE_VERIFY_OTP_BUTTON)}
```
Comment (ArthurO2002) [inline src/pages/reset-password/verify-otp.tsx:32]: Loading

**Review by ArthurO2002 — CHANGES_REQUESTED** [review]

File: src/pages/register/index.tsx [hunk 7]
```diff
@@ -107,7 +108,11 @@ export const Register = () => {
               size="fl"
               type="submit"
             >
-              {t(Translation.PAGE_REGISTER_INPUT_ACTIONS_CREATE_ACCOUNT)}
+              {isSignUpLoading ? (
+                <IsLoading />
+              ) : (
+                t(Translation.PAGE_REGISTER_INPUT_ACTIONS_CREATE_ACCOUNT)
+              )}
```
[the remark this answers is not in the document]
Comment (DavMovsesyan2002) [reply src/pages/register/index.tsx:115]: Done

File: src/manager/authentication/authManager.ts [hunk 8]
```diff
@@ -20,8 +23,28 @@ const authManager = {
...
+  resetPassword(params: IResetPasswordParams) {
+    return axiosInstance.post<null, IAxiosResponse<null>>(
+      `${baseURL}/v1/auth/reset-password`,
+      params
+    )
+  },
   googleSignIn() {
-    return axiosInstance.post<null, IAxiosResponse<null>>(`${baseURL}/v1/auth/google`)
+    return axiosInstance.get<null, IAxiosResponse<null>>(
+      `http://localhost:2002/users/v1/auth/google`
```
Comment (ArthurO2002) [inline src/manager/authentication/authManager.ts:47]: Never keep this kind of hardcoded data

**Review by ArthurO2002 — CHANGES_REQUESTED** [review]
Button should have different loading and disabled states

**Pull request merged**